File: CloudMusic/src/main/java/com/ken/android/CloudMusic/Activity/BaseActivity.java
package com.ken.android.CloudMusic.Activity;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.Window;
import android.view.WindowManager;

import org.xutils.x;

/**
 * Created by axnshy on 16/8/9.
 */
public class BaseActivity extends AppCompatActivity{
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        x.view().inject(this);
        Window window = getWindow();
        // Translucent status bar
        window.setFlags(
                WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS,
                WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
    }
}

File: SmartPlay/src/main/java/com/luffy/smartplay/ui/base/BaseActivity.kt
package com.luffy.smartplay.ui.base

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModel
import androidx.viewbinding.ViewBinding

/**
 * Created by axnshy on 16/8/9.
 */
abstract class BaseActivity<T : ViewBinding,M:ViewModel> : BasePlayerActivity() {

    lateinit var viewBinding : T
    abstract val viewModel : M

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(bindViewBinding().apply {
            viewBinding = this
        }.root)
    }

    abstract fun bindViewBinding() : T
}